### ISSUE TITLE
Robustness to horizontal flipping

### ISSUE THREAD

**Issue by benemana** [Peachypie98/RivaGAN#1]
Hi, thank you for this amazing work of refactoring the original source code of RivaGAN.

I was wondering if you have any suggestion on how to give more robustness to the model against horizontal flipping attacks.

I trained the model, but after some experiments I noticed that flipping the video leads to very poor accuracy in watermark extraction.

I tried to implement a simple noise layer like this:

```
class HorizontalFlipping(nn.Module):
    """
    Simulates a transformation where the input is mirrored horizontally.

    Input: (N, 3, L, H, W)
    Output: (N, 3, L, H, W)
    """

    def __init__(self):
        super(HorizontalFlipping, self).__init__()

    def forward(self, frames):
        #return frames.flip(dims=(-1,))
        return frames.flip([4]) #flips along the width dimension
```

and then I added it inside the training and validation code, similarly to the others noise layers already implemented (Crop, Compression, Scale).

The problem is, when I run a new training with such modifications, the various accuracies have a worsening (included the validation crop accuracy, scale accuracy, ...).

Do you have any suggestions?
Thanks a lot

**Comment by Peachypie98** [comment]
Apologies for the delayed reply. However, it appears that you have already resolved the issue.
